app/assets/src/components/ui/controls/dropdowns/common/DropdownTrigger.jsx

```diff
@@ -17,6 +17,8 @@ class DropdownTrigger extends React.Component {
       placeholder,
     } = this.props;
 
+    const hasBadgeCount =
+      label === "Threshold Filters:" || label === "Categories:";
```

Comment:
A hard-coded conditional like this is error prone. If someone adds another dropdown that doesn't have label Threshold Filters or Categories, this code won't work for it and the bug will re-appear. The chance of that happening is quite high.

It looks like the cases we want to catch are ones where `<DropdownLabel>` is passed in as `this.props.value`. A better approach would be to check if `value` is a `DropdownLabel` component:
```
const hasDropdownLabel = get('type.name', this.props.value) === "DropdownLabel".
```
We use `get` from `lodash/fp` instead of doing `this.props.value.type.name` because if `value` is a string this second expression will fail.

I could add a comment explaining why we're taking this approach.
"When a DropdownLabel is passed in as the value, we need to add special styling so that the DropdownTrigger displays properly for narrow widths."

Reply:
This is great feedback! Thank you!! @MarkAZhang